OpenSim/Simulation/StatesTrajectory.cpp

```diff
@@ -108,6 +108,60 @@ bool StatesTrajectory::isCompatibleWith(const Model& model) const {
     return true;
 }
 
+// Hide this function from other translation units.
+namespace {
+    template <typename T>
+    std::vector<std::string> createVector(const T& strings) {
+        std::vector<std::string> vec;
+        for (int i = 0; i < strings.size(); ++i)
+            vec.push_back(strings[i]);
+        return vec;
+    }
+    template <>
+    std::vector<std::string> createVector(
+            const std::vector<std::string>& strings) {
+        return strings;
+    }
+}
+
+TimeSeriesTable StatesTrajectory::exportToTable(const Model& model,
+        const std::vector<std::string>& requestedStateVars) const {
+
+    OPENSIM_THROW_IF(!isCompatibleWith(model),
+                     StatesTrajectory::IncompatibleModel, model);
+
+    // This code is based on DelimFileAdapter::extendRead().
+    TimeSeriesTable table;
+
+    // Set the column labels as metadata.
+    std::vector<std::string> stateVars = requestedStateVars.empty() ?
+            createVector(model.getStateVariableNames()) :
+            createVector(requestedStateVars);
+    ValueArray<std::string> valueArray;
+    for (const auto& labels : stateVars) {
+        valueArray.upd().push_back(SimTK::Value<std::string>(labels));
+    }
+    int numDepColumns = valueArray.size();
+    TimeSeriesTable::DependentsMetaData depMetadata;
+    depMetadata.setValueArrayForKey("labels", valueArray);
+    table.setDependentsMetaData(depMetadata);
+
```

Comment:
This code(Lines 140 - 147 ) to set column labels is ugly (due to the inconvenient/infexible DataTable interface). However, it is now possible to avoid this by using the method `table.setColumnLabels()` you suggested I add to DataTable.